cob_obstacle_distance/include/cob_obstacle_distance/marker_shapes.hpp

```diff
@@ -52,30 +51,30 @@ class MarkerShape : public IMarkerShape
         FclMarkerConverter<T> fcl_marker_converter_;
         visualization_msgs::Marker marker_;
 
-        void init(double x, double y, double z,
+        void init(const std::string root_frame, double x, double y, double z,
                   double quat_x, double quat_y, double quat_z, double quat_w,
                   double color_r, double color_g, double color_b, double color_a);
 
     public:
-        MarkerShape(T &fcl_object, geometry_msgs::Pose &pose, std_msgs::ColorRGBA &col)
-        : MarkerShape(fcl_object,
+        MarkerShape(const std::string root_frame, T& fcl_object, geometry_msgs::Pose& pose, std_msgs::ColorRGBA& col)
+        : MarkerShape(root_frame, fcl_object,
                 pose.position.x, pose.position.y, pose.position.z,
                 pose.orientation.x, pose.orientation.y, pose.orientation.z, pose.orientation.w,
                 col.r, col.g, col.b, col.a)
         {
         }
 
-        MarkerShape(T &fcl_object, geometry_msgs::Point &pos, geometry_msgs::Quaternion &quat, std_msgs::ColorRGBA &col)
-        : MarkerShape(fcl_object, pos.x, pos.y, pos.z, quat.x, quat.y, quat.z, quat.w, col.r, col.g, col.b, col.a)
```

Comment:
As we started this, should we check all packages/code files for any `class& name` vs. `class &name` inconsistencies? At least in `cob_twist_controller` and `cob_obstacle_distance`...
(same for  `class* name` vs. `class *name` in case this occurs somewhere)